src/interfaces/wasi/wit_generation.rs

```diff
@@ -0,0 +1,269 @@
+// Since wit_generation.rs code is called from the Cargo feature `wasi_impl`,
+// rustc cannot see that code is actually not a dead code
+#![allow(dead_code)]
+
+use cosmian_crypto_base::{
+    hybrid_crypto::Metadata, symmetric_crypto::aes_256_gcm_pure::Aes256GcmCrypto,
+};
+use witgen::witgen;
+
+use crate::{
+    core::{
+        bilinear_map::bls12_381::Bls12_381,
+        gpsw::{AbeScheme, AsBytes, Gpsw},
+        policy::{attr, AccessPolicy},
+        Engine,
+    },
+    interfaces::hybrid_crypto::{
+        decrypt_hybrid_block, decrypt_hybrid_header, encrypt_hybrid_block,
+    },
+};
+
+type PublicKey = <Gpsw<Bls12_381> as AbeScheme>::MasterPublicKey;
+type PrivateKey = <Gpsw<Bls12_381> as AbeScheme>::MasterPrivateKey;
+type UserDecryptionKey = <Gpsw<Bls12_381> as AbeScheme>::UserDecryptionKey;
+type DelegationKey = <Gpsw<Bls12_381> as AbeScheme>::MasterPublicDelegationKey;
+
+#[witgen]
+pub struct PolicyAxis {
+    pub name: String,
+    pub attributes: Vec<String>,
+    pub hierarchical: bool,
+}
+
+/// This struct only provides a visual way to display policy arguments
+#[witgen]
+pub struct Policy {
+    pub primary_axis: PolicyAxis,
+    pub secondary_axis: PolicyAxis,
+}
+
+/// Regroup private, public and delegation keys in same struct
+#[witgen]
+pub struct MasterKey {
+    pub private_key: Vec<u8>,
+    pub public_key: Vec<u8>,
+    pub delegation_key: Vec<u8>,
+    pub policy_serialized: Vec<u8>,
+}
+
+/// This struct only provides a visual way to display attributes arguments
+#[witgen]
+pub struct Attribute {
+    pub axis_name: String,
+    pub attribute: String,
+}
+impl Attribute {
+    fn abe_attributes(attributes: Vec<Attribute>) -> Vec<crate::core::policy::Attribute> {
```

Comment:
take by ref? or is it a bindgen facility maybe? (others later on)
```suggestion
    fn abe_attributes(attributes: &[Attribute]) -> Vec<crate::core::policy::Attribute> {
```